frontend/src/components/Shared/CrossingMapPage/CrossingMapPage.js

```diff
@@ -149,12 +139,45 @@ class CrossingMapPage extends Component {
       selectedCommunity,
     } = this.state;
     const { currentUser } = this.props;
-    const allCommunities =
-      this.props.data &&
-      !this.props.data.loading &&
-      this.props.data.allCommunities
-        ? this.props.data.allCommunities.nodes
-        : null;
+
+    const isLoading =
+      !this.props.openCrossings ||
+      this.props.openCrossings.loading ||
+      !this.props.closedCrossings ||
+      this.props.closedCrossings.loading ||
+      !this.props.cautionCrossings ||
+      this.props.cautionCrossings.loading ||
+      !this.props.longtermCrossings ||
+      this.props.longtermCrossings.loading ||
+      !this.props.allCommunities ||
+      this.props.allCommunities.loading;
+
+    if (
+      !isLoading &&
+      (this.props.openCrossings.searchCrossings == null ||
+        this.props.closedCrossings.searchCrossings == null ||
+        this.props.cautionCrossings.searchCrossings == null ||
+        this.props.longtermCrossings.searchCrossings == null)
+    ) {
+      // TODO: add error logging
+      return <div>Error Loading</div>;
+    }
+
+    const allCommunities = !isLoading
+      ? this.props.allCommunities.allCommunities.nodes
+      : null;
```

Comment:
It's a little more clear and terse to write:

```
!isLoading && this.props.allCommunities.allCommunities.nodes
```

Instead of

```
!isLoading
  ? this.props.allCommunities.allCommunities.nodes
  : null
```

Reply:
When `isLoading` is `false`, `!isLoading && this.props.allCommunities.allCommunities.nodes` will return `false` while 
```
!isLoading
  ? this.props.allCommunities.allCommunities.nodes
  : null
```
will return `null`

Reply:
👍 Good point.